client/battle/BattleStacksController.cpp

```diff
@@ -321,7 +321,7 @@ void BattleStacksController::showStackAmountBox(Canvas & canvas, const CStack *
 	{
 		double healthMaxType = stack->unitType()->getMaxHealth();
 		double healthMaxStack = stack->getMaxHealth();
-		double healthMaxRatio = healthMaxStack / healthMaxType;
+		double healthMaxRatio = std::min(healthMaxStack / healthMaxType, 1.0);
```

Comment:
Wouldn't such a fix lead to a similar kind of bug - where a half-dead unit still has a full health bar?